binder/requirements.txt

```diff
@@ -1,22 +1,3 @@
-dask>=0.15.2
-distributed>=1.18.3
-future>=0.16.0
-ipaddress>=1.0.18; python_version <= '2.7'
-ipython>=5.3.0
-matplotlib>=2.0.0
-mock>=2.0.0
-notebook>=4.4.1
-numpy>=1.10.4
-pandas-datareader>=0.5.0
-pandas>=0.20.3
-patsy>=0.4.1
-pytest-cov>=2.3.1
-pytest-xdist>=1.15.0
-pytest>=3.0.2
-requests>=2.9.1
-scikit-learn>=0.17.1
-scipy>=0.17.0
-seaborn>=0.7.1
-six>=1.10.0
-statsmodels>=0.6.1
-tqdm>=4.10.0
+-e .  # relative to cwd
+-r ../requirements.txt  # relative to here
+-r ../test-requirements.txt  # relative to here
```

Comment:
Isn't the order skewed? I think you should install requirements first, then test-requirements and finally the package.

further for the package, shouldn't you have -e ../. as well?

Reply:
On the last point, no; pip is broken like that: https://github.com/pypa/pip/issues/3772

Reply:
On the order, it doesn't really matter, but I think you want any specific version in the requirements.txt files to override the generic ones in the package.